packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js

```diff
@@ -35,21 +36,25 @@ export default function usePatternDetails( postType, postId ) {
 			select( editorStore ).__experimentalGetDefaultTemplatePartAreas(),
 		[]
 	);
-	const { currentTheme, patternCategories } = useSelect( ( select ) => {
+	const { currentTheme, userPatternCategories } = useSelect( ( select ) => {
 		const { getCurrentTheme, getUserPatternCategories } =
 			select( coreStore );
-		const userPatternCategories = getUserPatternCategories();
-		const categories = new Map();
-		userPatternCategories.forEach( ( userCategory ) =>
-			categories.set( userCategory.id, userCategory )
-		);
 
 		return {
 			currentTheme: getCurrentTheme(),
-			patternCategories: categories,
+			userPatternCategories: getUserPatternCategories(),
 		};
 	}, [] );
 
+	const patternCategories = useMemo( () => {
+		const categories = new Map();
+		userPatternCategories.forEach( ( userCategory ) =>
+			categories.set( userCategory.id, userCategory )
+		);
+
+		return categories;
+	}, [ userPatternCategories ] );
```

Comment:
I think we can skip `useMemo` here. The value is only needed for the `details` array, which isn't used in effect or by the memoized component. What do you think?

Reply:
Fair enough. I've removed the `useMemo` use and relocated the creation of the map to where it's actually used.